src/main/scala/it/pagopa/interop/authorizationprocess/server/impl/Main.scala

```diff
@@ -88,8 +91,12 @@ object Main extends App with Dependencies {
   def updateKeys(key: Parameter): Future[Unit] = {
     logger.info(s"Update keys for client ${key.clientId}")
     for {
-      relationship <- partyManagementService.getRelationshipById(key.relationShipId)
-      _            <- authorizationManagementService.updateKey(key.clientId, key.kid, relationship.from)
+      relationship  <- partyManagementService.getRelationshipById(key.relationShipId)
+      persistentKey <- authorizationManagementService.getClientKey(key.clientId, key.kid)
+      _             <- persistentKey.userId match {
+        case Some(userId) if (userId == key.relationShipId) => Future.unit
+        case _ => authorizationManagementService.migrateKeyRelationshipToUser(key.clientId, key.kid, relationship.from)
+      }
```

Comment:
the `userId` should never be equal to `relationShipId`.

Btw an easy way to implement this behaviour is to leave this as it was
```suggestion
      _ <- authorizationManagementService.updateKey(key.clientId, key.kid, relationship.from)
```

and change [this](https://github.com/pagopa/interop-be-authorization-process/pull/229/files/5d3ece7c0a9ec513b7a3338b0e9acf1bc6b47d53..db9f97b8722021d294279313f2cdb373820a8654#diff-5f7f8be5e71cb7c899d3195a3b6437393f3574794445e5db00a1d252f0bc6a07R65) to
```
client.keys.collect { case PersistentKey(Some(relationshipId), None, kid, _, _, _, _, _) =>
```
this way we can work only on non-migrated keys